docs/spec/distributing.rst

```diff
@@ -10,72 +10,428 @@ Stub files
+in the ``typing`` module, plus these additional ones:
+
+ * ``classmethod``
+ * ``staticmethod``
+ * ``property`` (including ``.setter``)
+ * ``abc.abstractmethod``
+ * ``dataclasses.dataclass``
+ * functions decorated with ``@typing.dataclass_transform``
+
+The behavior of other decorators should instead be incorporated into the types.
+For example, for the following function::
+
+  import contextlib
+  @contextlib.contextmanager
+  def f():
+      yield 42
+
+the stub definition should be::
+
+  from contextlib import AbstractContextManager
+  def f() -> AbstractContextManager[int]: ...
+
+Version and Platform Checks
+"""""""""""""""""""""""""""
+
+Stub files for libraries that support multiple Python versions can use version
+checks to supply version-specific type hints. Stubs for different Python
+versions should still conform to the most recent supported Python version's
+syntax, as explain in the :ref:`stub-file-syntax` section above.
+
+Version checks are if-statements that use ``sys.version_info`` to determine the
+current Python version. Version checks should only check against the ``major`` and
+``minor`` parts of ``sys.version_info``. Type checkers are only required to
+support the tuple-based version check syntax::
+
+    if sys.version_info >= (3,):
+        # Python 3-specific type hints. This tuple-based syntax is recommended.
+    else:
+        # Python 2-specific type hints.
+
+    if sys.version_info >= (3, 5):
+        # Specific minor version features can be easily checked with tuples.
+
+    if sys.version_info < (3,):
+        # This is only necessary when a feature has no Python 3 equivalent.
+
+Stubs should avoid checking against ``sys.version_info.major`` directly and
+should not use comparison operators other than ``<`` and ``>=``.
+
+No::
+
+    if sys.version_info.major >= 3:
+        # Semantically the same as the first tuple check.
+
+    if sys.version_info[0] >= 3:
+        # This is also the same.
+
+    if sys.version_info <= (2, 7):
+        # This does not work because e.g. (2, 7, 1) > (2, 7).
+
+Some stubs also may need to specify type hints for different platforms. Platform
+checks must be equality comparisons between ``sys.platform`` and the name of a
+platform as a string literal:
+
+Yes::
+
+    if sys.platform == 'win32':
+        # Windows-specific type hints.
+    else:
+        # Posix-specific type hints.
+
+No::
+
+    if sys.platform.startswith('linux'):
+        # Not necessary since Python 3.3.
+
+    if sys.platform in ['linux', 'cygwin', 'darwin']:
+        # Only '==' or '!=' should be used in platform checks.
+
+Version and platform comparisons can be chained using the ``and`` and ``or``
+operators::
+
+    if sys.platform == 'linux' and (sys.version_info < (3,) or sys,version_info >= (3, 7)): ...
+
+Enums
+"""""
+
+Enum classes are supported in stubs, regardless of the Python version targeted by
+the stubs.
+
+Enum members should be specified with an unannotated assignment, for example as
```

Comment:
This is already covered in the "Enums" section of the typing spec and should not be duplicated here.